x-pack/solutions/observability/plugins/apm/public/components/app/onboarding/instructions/otel_agent.tsx

```
@@ -193,7 +193,7 @@ export function OpenTelemetryInstructions({
     {
       setting: 'OTEL_RESOURCE_ATTRIBUTES',
       value:
-        'service.name=<app-name>,service.version=<app-version>,deployment.environment=production',
+        'service.name=<app-name>,service.version=<app-version>,deployment.environment.name=production',
```

Should this match the SemConv version of the SDK which the user configures uses? This change was introduced in Semconv [v1.27.0](https://github.com/open-telemetry/semantic-conventions/releases/tag/v1.27.0)

Context:
Currently we are handling only the old version in the stack - it means as of today, only `deployment.environment` is used properly, `deployment.environment.name` is basically ignored. We should handle both (issue to be linked here).

But even if we handle both - if the SDK uses a SemConv pre 1.27, then `deployment.environment` is the matching name - otherwise the SDK will mix versions.

Overall, I'm not sure it's a good idea to merge it today without full support for `deployment.environment.name`  in the whole stack. So even if users use post 1.27 version - today setting  `deployment.environment` is the one which will work.

So 2 things I raise here:
- Right now only  `deployment.environment` will be used by the stack and I'd wait with the merge until the new one is supported
- (the less relevant comment) If the new is supported - should this maybe still depend on the SemConv version of the SDK and not just generally suggest the new version which may not be correct to the specific SDK?

---

I agree with you on the fact that we can't merge it as-is for now until we have a better support for both attributes.

For supporting the new attribute, ideally this could be something dependent on the SDK version, however while we might have the SDK version, as far as I know we don't have any clue of the semconv version that is used by a particular SDK (maybe we could build and maintain our own mapping of this).

In practice though, this semconv resource attribute is passed "as-is" to the SDK, there is no meaning nor behavior in the SDK that depends on it. This is also the reason why the rename in 1.27.0 was a "silent failure" as the semconv update did not trigger any code changes in the EDOT Java agent because the related constant was not used at all.

In the mean time, does it means it would be better if EDOT agents ensure that both value are sent when possible ?

---

Opened https://github.com/elastic/opentelemetry-dev/issues/814 to track the support of the new field name.

>  however while we might have the SDK version, as far as I know we don't have any clue of the semconv version that is used by a particular SDK (maybe we could build and maintain our own mapping of this).

I agree with you - it's hard, if not impossible to know which SemConv version the thing that sends the data uses.

Probably over time this is not a huge issue - and since this is as you say just passed through as is, maybe just statically suggesting the newer name is ok - assuming we have general support for that in the stack.

> In the mean time, does it means it would be better if EDOT agents ensure that both value are sent when possible ?

Discussed in slack. My opinion is: yes, that's the safest way and that'd fix the problem for currently released versions of the EDOT collector and the stack. And then mid/long term the workaround may be removed if we have a general solution in  https://github.com/elastic/opentelemetry-dev/issues/814.